tests/unittests/tests-ecc: segfault with golay, when enabling asserts

<!--
====================================
IF YOUR ISSUE IS RELATED TO SECURITY
====================================
please submit it to the security mailing-list security@riot-os.org.

If your issue is a question related to the usage of RIOT, please submit it to
the user mailing-list users@riot-os.org or to the developer mailing-list
devel@riot-os.org.
-->

#### Description
<!--
Example: Cannot build gnrc_networking application for samr21-xpro board.
-->

tests/unittests/tests-ecc fails an assertion when compiled with `DEVELHELP=1`, somewhere in the golay code. DEVELHELP enables golay's assertions, which trigger on the test's input.

#### Steps to reproduce the issue

    $ DEVELHELP=1 make -C tests/unittests tests-ecc test
<!--
Try to describe as precisely as possible here the steps required to reproduce
the issue. Here you can also describe your hardware configuration, the network
setup, etc.
-->

#### Expected results
<!--
Example: The gnrc_networking application builds on samr21-xpro.
-->

Test passes.
#### Actual results
<!--
Please paste or specifically describe the actual output.
-->

```
[kaspar@ng riot.tmp (master)]$ DEVELHELP=1 make -C tests/unittests tests-ecc test                                                
make: Entering directory '/home/kaspar/src/riot.tmp/tests/unittests'                                                             
Building application "tests_unittests" for "native" with MCU "native".                                                           
                                                                
"make" -C /home/kaspar/src/riot.tmp/boards/native        
"make" -C /home/kaspar/src/riot.tmp/boards/native/drivers
"make" -C /home/kaspar/src/riot.tmp/core                                                                                         
"make" -C /home/kaspar/src/riot.tmp/cpu/native             
"make" -C /home/kaspar/src/riot.tmp/cpu/native/periph                                                                            
"make" -C /home/kaspar/src/riot.tmp/cpu/native/stdio_native                                                                      
"make" -C /home/kaspar/src/riot.tmp/cpu/native/vfs                                                                               
"make" -C /home/kaspar/src/riot.tmp/drivers                                                                                      
"make" -C /home/kaspar/src/riot.tmp/drivers/periph_common
"make" -C /home/kaspar/src/riot.tmp/sys
"make" -C /home/kaspar/src/riot.tmp/sys/ecc
"make" -C /home/kaspar/src/riot.tmp/sys/embunit
"make" -C /home/kaspar/src/riot.tmp/sys/test_utils/interactive_sync
"make" -C /home/kaspar/src/riot.tmp/tests/unittests/tests-ecc
   text    data     bss     dec     hex filename
  39776     831   47768   88375   15937 /home/kaspar/src/riot.tmp/tests/unittests/bin/native/tests_unittests.elf
r
/home/kaspar/src/riot.tmp/tests/unittests/bin/native/tests_unittests.elf  
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

main(): This is RIOT! (Version: 2020.04-devel-456-gca5ed3)
Help: Press s to start test, r to print it is ready
READY
s
START
...sys/ecc/golay2412.c:149 => 0x5660fbe5
*** RIOT kernel panic:
FAILED ASSERTION.

*** halted.


native: exiting
Unexpected end of file in expect script at "child.expect(u"OK \\([0-9]+ tests\\)")" (tests/unittests/tests/01-run.py:14)

Process already stopped
make: *** [/home/kaspar/src/riot.tmp/tests/unittests/../../Makefile.include:734: test] Error 1
make: Leaving directory '/home/kaspar/src/riot.tmp/tests/unittests'
[kaspar@ng riot.tmp (master)]$
```
#### Versions
<!--
Operating system: Mac OSX, Linux, Vagrant VM
Build environment: GCC, CLang versions (you can run the following command from
the RIOT base directory: make print-versions).
-->

<!-- Thanks for contributing! -->
https://gist.github.com/e2e40340ae354d13d9e81844a7dac65c

--- a/sys/ecc/golay2412.c
+++ b/sys/ecc/golay2412.c
@@ -146,7 +146,7 @@ static uint32_t golay2412_matrix_mul(uint32_t _v,
 static uint32_t golay2412_encode_symbol(uint32_t _sym_dec, const uint32_t *_A)
 {
     /* validate input */
-    assert(_sym_dec > (1 << 12));
+    assert(_sym_dec < (1 << 12));
 
     /* compute encoded/transmitted message: v = m*G */
     return golay2412_matrix_mul(_sym_dec, _A, 24);

